OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -393,16 +393,23 @@ public virtual FeatureDecision GetVariationForFeature(FeatureFlag featureFlag, s
         /// </summary>
         /// <param name = "userId" >User Identifier</param>
         /// <param name = "filteredAttributes" >The user's attributes.</param>
-        /// <returns>User ID if bucketing Id is not provided in user attributes, bucketing Id otherwise.</returns>
+        /// <returns>Bucketing Id if it is a string type in attributes, user Id otherwise.</returns>
         private string GetBucketingId(string userId, UserAttributes filteredAttributes)
         {
             string bucketingId = userId;
 
             // If the bucketing ID key is defined in attributes, then use that in place of the userID for the murmur hash key
             if (filteredAttributes != null && filteredAttributes.ContainsKey(ControlAttributes.BUCKETING_ID_ATTRIBUTE))
             {
-                bucketingId = filteredAttributes[ControlAttributes.BUCKETING_ID_ATTRIBUTE];
-                Logger.Log(LogLevel.DEBUG, string.Format("Setting the bucketing ID to \"{0}\"", bucketingId));
+                if (filteredAttributes[ControlAttributes.BUCKETING_ID_ATTRIBUTE] is string)
```

Comment:
Please combine these two lines. 

(filteredAttributes[ControlAttributes.BUCKETING_ID_ATTRIBUTE] as string ?? userId).ToString()
should serve the purpose, please check.

Reply:
leave it, as is used for casting not for type checking.